src/main/scala/no/ndla/listingapi/service/Auth.scala

```diff
@@ -0,0 +1,43 @@
+/*
+ * Part of NDLA listing_api.
+ * Copyright (C) 2017 NDLA
+ *
+ * See LICENSE
+ */
+
+package no.ndla.listingapi.service
+
+import no.ndla.listingapi.model.api.AccessDeniedException
+import no.ndla.network.AuthUser
+
+trait AuthenticationRole {
+
+  val authRole: AuthRole
+
+  class AuthRole {
+    def assertHasRole(role: String): Unit = {
+      if (!AuthUser.hasRole(role))
+        throw new AccessDeniedException("User is missing required role to perform this operation")
+    }
+  }
+
+}
+
+trait AuthenticationUser {
+
+  val authUser: AuthUser
+
+  class AuthUser {
+
+    def id(): String = {
+      if (AuthUser.get.isEmpty || AuthUser.get.get.isEmpty) {
+        throw new AccessDeniedException(("User id required to perform this operation"))
+      } else {
+        return AuthUser.get.get
+      }
+
+    }
+
+  }
+
+}
```

Comment:
Kan godt splitte fila i to hvis du syns det kommuniserer bedre. Jeg la dem i samme for å nettopp samle auth ting i samme fil.